### PR TITLE
Update Project Exporting

### PR DESCRIPTION
**Pull request by andreiborodin**
This is the continuation of the piece-wise update of the exporting logic, to restructure to a cleaner seperation of responsibilites.

### CODE REVIEW

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/SDKProjectTemplate.g.props.template [hunk 1]
```diff
@@ -9,9 +9,8 @@
     <ProjectGuid><!--PROJECT_GUID_TOKEN--></ProjectGuid>
     <AllowUnsafeBlocks><!--ALLOW_UNSAFE_TOKEN--></AllowUnsafeBlocks>
     <LangVersion><!--LANGUAGE_VERSION_TOKEN--></LangVersion>
-    <DevelopmentBuild><!--DEVELOPMENT_BUILD_TOKEN--></DevelopmentBuild>
+    <DevelopmentBuild>false</DevelopmentBuild>
```
Comment (keveleigh) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/SDKProjectTemplate.g.props.template]: Intentionally hardcoded here?

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/IUnityProjectExporter.cs [hunk 2]
```diff
@@ -67,6 +60,8 @@ public interface IUnityProjectExporter
         /// <param name="path">The <see cref="FileInfo"/> representing where this props file will be written.</param>
         /// <param name="scriptingBackend">The scripting backend for the platform props.</param>
         IWSAPlayerPlatformPropsExporter CreateWSAPlayerPlatformPropsExporter(FileInfo path, ScriptingBackend scriptingBackend);
+
+        ICSharpProjectExporter CreateCSharpProjectExporter(FileInfo filePath, DirectoryInfo generatedProjectFolder, bool isGenerated);
```
Comment (keveleigh) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/IUnityProjectExporter.cs]: Add a summary, to match the other methods in this interface?

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/SDKProjectTemplate.g.props.template (hunk 1 above)
Comment (andreiborodin) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/SDKProjectTemplate.g.props.template]: Yes, as several other properties. I never actually set this to anything other than false, and override it in certain conditions in this project.

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/IUnityProjectExporter.cs (hunk 2 above)
Comment (andreiborodin) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/IUnityProjectExporter.cs]: Ah yeah, good call!

**Pull request merged**